st2client/st2client/commands/auth.py

```diff
@@ -176,39 +177,37 @@ def __init__(self, resource, *args, **kwargs):
 
         super(WhoamiCommand, self).__init__(
             resource, kwargs.pop('name', 'create'),
-            'Display the currently authenticated/configured user',
+            'Display the currently authenticated user',
             *args, **kwargs)
 
     def run(self, args, **kwargs):
+        user_info = self.app.client.get_user_info(**kwargs)
+        return user_info
 
-        cli = BaseCLIApp()
-
-        # Determine path to config file
+    def run_and_print(self, args, **kwargs):
         try:
-            config_file = cli._get_config_file_path(args)
-        except ValueError:
-            # config file not found in args or in env, defaulting
-            config_file = config_parser.ST2_CONFIG_PATH
-
-        # Update existing configuration with new credentials
-        config = ConfigParser()
-        config.read(config_file)
+            user_info = self.run(args, **kwargs)
+        except Exception as e:
+            response = getattr(e, 'response', None)
+            status_code = getattr(response, 'status_code', None)
+            is_unathorized_error = (status_code == httplib.UNAUTHORIZED)
 
-        return config.get('credentials', 'username')
+            if response and is_unathorized_error:
+                print('Not authenticated')
+            else:
+                print('Unable to retrieve currently logged-in user')
 
-    def run_and_print(self, args, **kwargs):
-        try:
-            username = self.run(args, **kwargs)
-            print("Currently logged in as %s" % username)
-        except KeyError:
-            print("No user is currently logged in")
-            if self.app.client.debug:
-                raise
-        except Exception:
-            print("Unable to retrieve currently logged-in user")
             if self.app.client.debug:
                 raise
 
+        print('Currently logged in as "%s".' % (user_info['username']))
+        print('')
+        print('Authentication method: %s' % (user_info['authentication']['method']))
```

Comment:
In subsequent pass, we could probably include expiry information for token. I am split because of the security concerns.

Reply:
Yeah, we can format this info better as well. If you have any idea on the format, let me know.

Reply:
There we go:

```bash
(virtualenv)vagrant@vagrant-ubuntu-trusty-64:/data/stanley$ python ./st2client/st2client/shell.py  whoami
Currently logged in as "testu".

Authentication method: authentication token
Authentication token expire time: 2017-11-14T10:26:35.724000Z

RBAC:
 - Enabled: False
 - Roles: 
```

We can also format the date nice aka more user-friendly.